compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingInterfaces.td

```diff
@@ -264,4 +264,80 @@ def IREEEncoding_EncodingTypeInterface :
   ];
 }
 
+//===----------------------------------------------------------------------===//
+// Encoding Propagation Attribute & Op Interfaces
+//===----------------------------------------------------------------------===//
+
+def IREEEncoding_PropagationAttrInterface :
+  AttrInterface<"EncodingPropagationAttrInterface"> {
+  let cppNamespace = "::mlir::iree_compiler::IREE::Encoding";
+  let description = [{
+    Interface used to query new encoding attributes that can be propagated to
+    the operands and result of the operation.
+  }];
+
+  let methods = [
+    InterfaceMethod<
+        [{
+          Returns true if the encodings can be propagated across the operation.
+        }],
+        /*retTy=*/"bool",
+        /*methodName=*/"isPropagable",
+        /*args=*/(ins
+          "::mlir::Value":$target
+        ),
+        /*methodBody=*/"",
+        /*defaultImplementation=*/[{
+            return false;
+        }]
+    >,
+    InterfaceMethod<
+        [{
+          Returns the new encodings for operand and result types
+          for the given operation.
```

Comment:
We don't have lint for td file, but I'd generally format this kind of description within 80 chars per line when possible. Maybe we can format it a bit in the first place.

```suggestion
          Returns the new encodings for operand and result types for the given
          operation.
```